source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -347,6 +351,27 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
   }
 
   parsed_alpn_protocols_ = parseAlpnProtocols(config.alpnProtocols());
+
+  // Ciphers
```

Comment:
How were these determined? I wonder if there is any sane way to prevent these from getting out of date? Can they be read from boringssl somehow?

Reply:
not sure if they’ll can be read from boringssl. I added this comment though:
```
  // Pre-populate a set of builtin stat names in a map that that is read in the
  // hot-path without a mutex. These were determined via logging while running
  // unit tests. If an additional cipher is added, the penalty is a potentially
  // mutex-contended addition to the set's dynamic map.
  //
  // TODO(jmarantz): Find a way to ensure we remember to update these names if
  // a new one is added. One option would be to keep an atomic counter of
  // dynamic lookups in StatNameSet and EXPECT that to be 0 in the test-class
  // destructor in SslContextImplTest.

```

Reply:
I've been messing around with this and have not found a satisfactory way to ensure these never get out of date.

The way I enumerated them initially was by adding a log-statement when they a stat was incremented and ran all tests. Then I collected the logs and ran `sort|uniq` before adding them to the code here.

Reply:
Reflecting on this more I am thinking that the challenge is broader than this one set of stats.  For example in http codes I made it a bit faster to report stats for status codes of 200, 404, and 503 but it would be useful to see if we are missing other frequently used codes.

Ditto for grpc methods.

I think we could build in some logging mechanism into SymbolTableImpl and StatNameSet where bursts of lookups for the same stat name could be logged as warnings, with a limit of one warning per (say) 10 minutes.

The message would include a list of the names being looked up and a shortlink to a doc page explaining how to get that name into the built-ins list, by source code edit or config.

This would be a separate pr obviously and should block #4980 but not this PR.  Wdyt?

There are still other comments on this PR I will take care of.

Reply:
Also we could potentially expose the frequently-looked-up stats in a new admin endpoint.

Reply:
IMO the ones I added are all reasonably obvious. The ones in this PR are the most subtle as I did not find a place in the code or a spec where these were enumerated. But how about I just doc what process I used to find them?